testing_app/README.md

```diff
@@ -0,0 +1,52 @@
+# testing_app
+
+A Sample app that shows different types of testing in Flutter.
+
+This particular sample uses the [Provider][] package but any other state management approach
+would do.
+
+[provider]: https://pub.dev/packages/provider
+
+## Goals for this sample
+
+Show how to perform:
+
+- Widget Testing,
+- Flutter Driver(Integration) Testing,
+- Performance Testing, and
+- State Management Testing using the [Provider][] package.
+
+## How to run tests
+- Navigate to the project's root folder using command line and follow the instructions below.
+
+### To run tests using only the Flutter SDK:
+The Flutter SDK can run some of the testing types on its own viz. unit testing and widget testing without the need of a physical device/emulator.
+- To run all the test files in the `test/` directory in one go, run `flutter test`.
+- To run a particular test file, run `flutter test test/<file_path>`
+
+### To run tests on a physical device/emulator:
+- Widget Tests:
+  - Run `flutter run test/<file_path>`
+- Flutter Driver Tests:
+  - Run `flutter drive --target=test_driver/<file_path>`
+  - eg. `flutter drive --target=test_driver/app.dart` to run the test in `test_driver/app_test.dart`
+- Performance Tests:
+  - Run `flutter drive --target=test_driver/<file_path> --profile` (Using a physical device and running performance tests in profile mode is recommended).
```

Comment:
The current sample seems to only have `perf.dart`, `perf_test.dart` for performance tests. Can we directly put `flutter drive --target=test_driver/perf.dart --profile` here to save some possible confusion of `flutter drive --target=test_driver/perf_test.dart --profile`?

Also, `app.dart` and `perf.dart` seem to be identical. You may be able to combine them into a single file as did in  https://github.com/flutter/flutter/pull/61621 (thanks @CareF). By doing that, your command will look like `flutter drive --target=test_driver/app.dart --driver test_driver/perf_test.dart`.

Reply:
I too was thinking that they're completely the same while writing the test but didn't know that we could use just one file for both of them. Thanks for sharing the reference. I've made the changes.